lib/block-supports/border.php

```diff
@@ -94,14 +94,18 @@ function gutenberg_apply_border_support( $block_type, $block_attributes ) {
 		! wp_should_skip_block_supports_serialization( $block_type, '__experimentalBorder', 'color' )
 	) {
 		$preset_border_color          = array_key_exists( 'borderColor', $block_attributes ) ? "var:preset|color|{$block_attributes['borderColor']}" : null;
-		$custom_border_color          = _wp_array_get( $block_attributes, array( 'style', 'border', 'color' ), null );
+		$custom_border_color          = isset( $block_attributes['style']['border']['color'] )
+			? _wp_array_get( $block_attributes, array( 'style', 'border', 'color' ), null )
+			: null;
 		$border_block_styles['color'] = $preset_border_color ? $preset_border_color : $custom_border_color;
 	}
 
 	// Generate styles for individual border sides.
 	if ( $has_border_color_support || $has_border_width_support ) {
 		foreach ( array( 'top', 'right', 'bottom', 'left' ) as $side ) {
-			$border                       = _wp_array_get( $block_attributes, array( 'style', 'border', $side ), null );
+			$border                       = isset( $block_attributes['style']['border'][ $side ] )
+				? _wp_array_get( $block_attributes, array( 'style', 'border', $side ), null )
```

Comment:
Same question as above here. Why not
```suggestion
				? $block_attributes['style']['border'][ $side ]
```